### PR TITLE
Qualified feature names

### LINKED ISSUE #212: Dropdowns need human-readable names

### PR DESCRIPTION
**Pull request by pierce-smith1**
Lots of internal changes that amount to the following things:
- Charts can now tell the difference between identically named constraints, i.e. things like SVI county and SVI tract are now separately plottable and will not be confused with one another.
- Human-readable names are now much easier to get (Closes #212).
- Scatterplots are now actually meaningful (guess what? Data coming into the scatterplots previously wasn't being joined on _anything_, meaning the distribution of data was effectively random. They are now properly joined by GISJOIN. Having to admit that I designed the world's most useless scatterplot was hard.)

Charting _may_ be noticeably slower with these changes, especially scatterplots.

### CODE REVIEW

**Review by honeybadgerofdoom — CHANGES_REQUESTED** [review]
Getting this error in the console when trying to change the x-axis or y-axis (scatterplots) via dropdown or toggles

scatterplotManager.js:132 Uncaught (in promise) TypeError: Cannot read property 'data' of undefined
    at scatterplotManager.js:132
    at Array.forEach (<anonymous>)
    at ScatterplotManager.prepareData (scatterplotManager.js:129)
    at ScatterplotManager.update (scatterplotManager.js:115)
    at chartSystem.js:139
    at Array.forEach (<anonymous>)
    at ChartSystem.update (chartSystem.js:139)

**Comment by drg101** [comment]
Same error for me on Chrome. Everything else looks super nice though.

**Comment by pierce-smith1** [comment]
Wouldn't you know it, this happens when you try to plot county vs. tract data because it can't join them together. For now I'll have this branch just disallow this type of plot and we'll figure out how this works with the UI later.

**Comment by pierce-smith1** [comment]
Scatterplots that try to mix county and tract data should now errorlessly self-destruct.

**Comment by drg101** [comment]
New error now: 
```
Uncaught (in promise) TypeError: Cannot read property 'type' of undefined
    at ScatterplotManager.prepareData (scatterplotManager.js:127)
    at ScatterplotManager.update (scatterplotManager.js:115)
    at chartSystem.js:139
    at Array.forEach (<anonymous>)
    at ChartSystem.update (chartSystem.js:139)
```

**Comment by drg101** [comment]
^ this happened when I tried to plot svi tract vs household income tract.

**Review by drg101 — CHANGES_REQUESTED** [review]
See above

**Comment by pierce-smith1** [comment]
I can't reliably reproduce this problem, but based on the error that's being thrown, this should fix it.

File: src/iframe/js/library/charting/scatterplotManager.js [hunk 1]
```diff
@@ -130,10 +130,15 @@ class ScatterplotManager {
...
-            data.push({
-                x: xfeat.find(d => d.GISJOIN === gisjoin).data,
-                y: yfeat.find(d => d.GISJOIN === gisjoin).data,
-            });
+            let xEntry = xfeat.find(d => d.GISJOIN === gisjoin);
+            let yEntry = yfeat.find(d => d.GISJOIN === gisjoin);
+            
+            if (xEntry && yEntry) {
+                data.push({
+                    x: xfeat.find(d => d.GISJOIN === gisjoin).data,
```
Comment (drg101) [inline src/iframe/js/library/charting/scatterplotManager.js:138]: Would be faster to do `xEntry.data` and `yEntry.data` so it doesn't have to do `.find` more times.

Comment (pierce-smith1) [reply src/iframe/js/library/charting/scatterplotManager.js]: Not sure how I didn't catch that, but it's changed now

**Review by drg101 — APPROVED** [review]
Good

**Review by honeybadgerofdoom — APPROVED** [review]
Looks good!

**Pull request merged**